runtime/lua/vim/lsp/util.lua

```diff
@@ -2207,6 +2207,50 @@ function M.lookup_section(settings, section)
   return settings
 end
 
+--- Helper function to resolve buffer number.
+--- Returns current buffer if bufnr is not valid
+---@param bufnr integer buffer number
+---@return integer buffer number
+function M.resolve_bufnr(bufnr)
+  return bufnr > 0 and bufnr or api.nvim_get_current_buf()
+end
+
+--- Request updated LSP information for a buffer.
+---
+---@param method string LSP method to call
+---@param opts (nil|table) Optional arguments
+---  - bufnr (integer, default: 0): Buffer to refresh
+---  - only_visible (boolean, default: false): Whether to only refresh for the visible regions of the buffer
+function M.refresh(method, opts)
```

Comment:
```suggestion

---@private
--- Request updated LSP information for a buffer.
---
---@param method string LSP method to call
---@param opts (nil|table) Optional arguments
---  - bufnr (integer, default: 0): Buffer to refresh
---  - only_visible (boolean, default: false): Whether to only refresh for the visible regions of the buffer
function M._refresh(method, opts)
```

I don't want to commit to this as public API. It's likely going to change with what I have in mind for a `vim.lsp.request`